### PR TITLE
Add Non-UVM support to preconditioner options

### PR DESCRIPTION
**Pull request by victorapm**
Removes the dependency on Unified Virtual Memory (UVM) for executing HMIS coarsening in BoomerAMG and ILU-threshold in ILU when Hypre is configured for GPUs. Although these algorithms continue to execute on the CPU, they can now run without requiring UVM support.

### CODE REVIEW

**Review by rfalgout — APPROVED** [review]
Looks good to me.  Thanks!

File: src/parcsr_ls/par_ilu_setup.c [hunk 1]
```diff
@@ -1775,42 +1764,31 @@ hypre_ILUSetupLDUtoCusparse(hypre_ParCSRMatrix  *L,
...
-      LDU_diag_data[pos++] = 1.0 / D[i - 1];
-      for (j = U_diag_i[i - 1]; j < U_diag_i[i]; j++)
+      for (j = U_diag_i[i]; j < U_diag_i[i + 1]; j++)
       {
-         LDU_diag_j[pos] = U_diag_j[j];
-         LDU_diag_data[pos++] = U_diag_data[j];
+         LDU_diag_j[pos]   = U_diag_j[j];
+         LDU_diag_a[pos++] = U_diag_a[j];
       }
    }
```
Comment (oseikuffuor1) [inline src/parcsr_ls/par_ilu_setup.c]: This seems to be resetting the diagonal to the first entry. Is there a reason for this other than to be consistent with some of the other situations in hypre? We may need to make sure it doesn't break any algorithmic assumptions downstream.

**Review by oseikuffuor1 — COMMENTED** [review]
@victorapm I have a question about the implications to doing solves on device if the UVM support is removed. Presumably, we would explicitly move data to device for the solve then. Let's discuss this at the meeting this afternoon

**Review by liruipeng — APPROVED** [review]
Looks good. Thank you @victorapm

**Review by oseikuffuor1 — APPROVED** [review]
Thanks Victor!

**Pull request merged**